lighthouse-cli/test/smokehouse/frontends/smokehouse-bin.js

```diff
@@ -132,20 +170,27 @@ async function begin() {
   const invertMatch = argv.invertMatch;
   const testDefns = getDefinitionsToRun(allTestDefns, requestedTestIds, {invertMatch});
 
-  const takeNetworkRequestUrls = () => {
-    return server.takeRequestUrls();
-  };
-
-  const options = {jobs, retries, isDebug: argv.debug, lighthouseRunner, takeNetworkRequestUrls};
-
   let isPassing;
+  let server;
+  let serverForOffline;
+  let takeNetworkRequestUrls = undefined;
+
   try {
-    server.listen(10200, 'localhost');
-    serverForOffline.listen(10503, 'localhost');
-    isPassing = (await runSmokehouse(testDefns, options)).success;
+    // If running the core tests, spin up the test server.
+    if (testDefnPath === coreTestDefnsPath) {
+      ({server, serverForOffline} = require('../../fixtures/static-server.js'));
```

Comment:
I think this is the first occurrence we've ever had of declarationless destructuring! 🎉

Reply:
oh you can do this?! Just need parentheses... nice.